pkg/analyzer/lib/file_system/physical_file_system.dart

```diff
@@ -101,15 +117,9 @@ class PhysicalResourceProvider implements ResourceProvider {
 
   @override
   Folder getStateLocation(String pluginId) {
-    String home;
-    if (io.Platform.isWindows) {
-      home = io.Platform.environment['LOCALAPPDATA'];
-    } else {
-      home = io.Platform.environment['HOME'];
-    }
-    if (home != null && io.FileSystemEntity.isDirectorySync(home)) {
```

Comment:
You have just lost checking for home folder existence.
I don't not whether it is important, but lets keep it for now.

Reply:
That check is in `_getHomePath`.